File: ETourBackend/src/main/java/com/example/demo/entity/Payment.java
package com.tourism.entity;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.math.BigDecimal;
import java.time.LocalDate;

@Entity
@Table(name = "payment")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Payment {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "payment_id")
    private Integer paymentId;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "booking_id")
    private Booking booking;

    @Column(name = "amount")
    private BigDecimal amount;

    @Column(name = "payment_date")
    private LocalDate paymentDate;

    @Column(name = "payment_status")
    private String paymentStatus;

    @Column(name = "payment_method")
    private String paymentMethod;
}

File: ETour-DotNet/Models/Payment.cs
namespace ETour.Api.Models;

public class Payment
{
    public int PaymentId { get; set; }
    public int? BookingId { get; set; }
    public Booking Booking { get; set; }
    public decimal? Amount { get; set; }
    public DateOnly? PaymentDate { get; set; }
    public string PaymentStatus { get; set; }
    public string PaymentMethod { get; set; }

    public string TransactionRef { get; set; }
}
